src/cci/cas_cci.c

```diff
@@ -712,25 +712,34 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
-    {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
-
-      get_last_error (con_handle, err_buf);
-      con_handle->used = false;
-      hm_release_connection (mapped_conn_id, &con_handle);
-    }
   else
     {
+      if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT])
+	{
+	  cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
+
+	  MUTEX_LOCK (con_handle_table_mutex);
+	  if (con_handle->id >= 0 && hm_put_con_to_pool (con_handle->id) >= 0)
```

Comment:
Valid values for con_handle->id are from 1 to 2048. ( Note new_con_handle_id())
if (con_handle->id >= 1 && hm_put_con_to_pool (con_handle->id) >= 0) 
or 
if ((con_handle->id >= 1 && con_handle->id <= MAX_CON_HANDLE) && hm_put_con_to_pool (con_handle->id ) >= 0)

Reply:
Good point, I agree.

Reply:
Thank you for your comment. It will be better than this.